Set TCP_NODELAY on for both client and server side

nemanja-boric-sociomantic opened the pull request:
Since the TCP_CORK flush is now deprecated and request implementations
should use explicit buffering, TCP_NODELAY is now always on.

Fixes #296

nemanja-boric-sociomantic committed:
Set TCP_NODELAY on for both client and server side

Since the TCP_CORK flush is now deprecated and request implementations
should use explicit buffering, TCP_NODELAY is now always on.

Fixes #296

nemanja-boric-sociomantic committed:
Remove deprecated flush_requested usage.

nemanja-boric-sociomantic commented:
I'm not sure if I should add fields for turning the  `TCP_NODELAY` _off_.

gavin-norman-sociomantic commented on src/swarm/neo/client/ConnectionSet.d:
```diff
@@ -47,13 +47,14 @@ public final class ConnectionSet : RequestOnConn.IConnectionGetter
     ***************************************************************************/
 
-    public bool socket_no_delay = false;
+    public bool socket_no_delay = true;
```
Should this be deprecated?

nemanja-boric-sociomantic replied:
Yes. This way we'll completely drop support for Nagle's algorithm, but I think that's fair enough.

gavin-norman-sociomantic replied:
Yeah, I think we just drop it for now. If we ever need to add it back, we can.

gavin-norman-sociomantic commented on src/swarm/neo/client/mixins/ClientCore.d:
```diff
@@ -972,6 +972,7 @@ template ClientCore ( )
 
     ***************************************************************************/
 
+    deprecated("This is now on by default, simply remove this call")
```
Release notes.

gavin-norman-sociomantic commented on src/swarm/node/model/NeoNode.d:
```diff
@@ -805,10 +802,12 @@ public class NodeBase ( ConnHandler : ISwarmConnectionHandler ) : INodeBase
         auto neo_conn_setup_params = new Neo.ConnectionHandler.SharedParams(
             options.epoll, options.shared_resources, options.requests,
-            options.no_delay, *credentials, this, &this.getResourceAcquirer);
+            no_delay, *credentials, this, &this.getResourceAcquirer);
```
Should the `SharedParams` field also be flagged in some way? e.g. a comment saying to remove it in the next major.

nemanja-boric-sociomantic replied:
Good point :+1:

gavin-norman-sociomantic replied:
Or maybe a deprecated ctor.

nemanja-boric-sociomantic replied:
Actually, I can rip out the fair chunk of this flag, just keeping the classes exposed to the *protos intact.

nemanja-boric-sociomantic replied:
I've pushed another commit ripping it all out.

nemanja-boric-sociomantic replied:
Looks like the interface is intact, it still compiles with the protos.

nemanja-boric-sociomantic committed:
Remove internal code for TCP_CORK/no_delay

TCP_NODELAY is now always on, so the internal code for flushing can be
completely removed from the swarm, just keeping the deprecated methods
exposed to the protos.

gavin-norman-sociomantic reviewed (commented):
Still needs deprecation release notes.

nemanja-boric-sociomantic committed:
Update deprecation release notes for the total removal of Nagle's algorithm

nemanja-boric-sociomantic commented:
Updated with the changed deprecation notes for the `TCP_CORK`.
